felix/daemon/daemon.go

```diff
@@ -504,6 +503,14 @@ configRetry:
 					break
 				}
 				log.WithError(err).Debug("Retrying Typha connection")
+
+				if !deleteWireguardCalled {
```

Comment:
This triggers deletion on the second attempt to connect which I think is a little crappy - typha could be restarting or something similar and in that case we don't just want to remove wireguard config.

I think we need to be a little smarter about detecting when our config is out of sync and therefore deletion would be a good thing to do.

Reply:
To that end we might want to  query the node that typha is on and find out its wireguard public key.  If we have the key programmed then wait longer, if not then do the Remove processing.